pulsar-client/src/test/java/org/apache/pulsar/client/impl/schema/PrimitiveSchemaTest.java

```diff
@@ -124,7 +124,7 @@ public void allSchemasShouldSupportNull(Map<Schema, List<Object>> testData) {
                 assertNull(((AbstractSchema) schema).decode(byteBuf),
                     "Should support null in " + schema.getSchemaInfo().getName() + " deserialization");
             } catch (NullPointerException npe) {
-                throw new NullPointerException("NPE when using schema " + schema + " : " + npe.getMessage());
+                throw new IllegalArgumentException("NPE when using schema " + schema + " : " + npe.getMessage(), npe);
```

Comment:
Why change NullPointerException to IllegalArgumentException here?

Reply:
`Avoid throwing NullPointerExceptions manually.` is the guide line I saw from codacy.